### PR TITLE
Candidate test for BES json log output

### PR DESCRIPTION
**Pull request by ndp-opendap**
1. Replaces the awk program `beslog2json.awk` with a python variant: `beslog2json.py` The python program is more robust. It should:
-  Eliminate printing non-json BS into the output stream.
-  Provide  limited  support for the log records with [] delimiters which seems like a miss but those kind of log lines have a way of sneaking into the log.
-  Turn problems with the log processing into legitimate json log errors
2. I added, to the `regression_test_script`, a test in which, after the Hyrax regression tests are run, the best.log file is retrieved. (commas FTW) The `beslog2json.py` code is run on the retrieved bes.log and the output is piped into `jg` If `jq` doesn't error then it's a win.

### CODE REVIEW

File: builds/ngap/beslog2json.py [hunk 1]
```diff
@@ -0,0 +1,560 @@
...
+# Global variable for debuggin control.
+debug_flag = False 
+
+##########################################################
+bes_log_field_delimiter = "|&|"
+bes_square_bracket_log_delimiter = "]["
+
+##########################################################
+# debug()
+# Writes "msg" to stderr when the debug_flag is enabled
```
Comment (Mikejmnez) [inline builds/ngap/beslog2json.py:62]: suggestion: include the definition of the function debug() within the scope of the function. For example:
```python
def debug(msg):
    """ writes `msg` to stderr when debug_flag is enabled
    """
```

File: builds/ngap/beslog2json.py [hunk 2]
```diff
@@ -0,0 +1,560 @@
...
+    if debug_flag: 
+        print("#", msg, file=sys.stderr)
+        sys.stderr.flush()
+
+
+##########################################################
+# torf()
+# Returns "true" (bool_val==True) or "false" (bool_val==false)
+#
+def torf(bool_val):
```
Comment (Mikejmnez) [inline builds/ngap/beslog2json.py:73]: same here (see comment above)

File: builds/ngap/beslog2json.py [hunk 3]
```diff
@@ -0,0 +1,560 @@
...
+RETURN_AS_KEY="return-as" # field 15
+LOCAL_PATH_KEY="local-path" # field 16 
+CE_KEY="constraint-expression" # field 17
+
+
+##########################################################
+# request_log_to_json()
+# Ingest a BES request log record.
+#
+def request_log_to_json(log_fields, json_log_line):
```
Comment (Mikejmnez) [inline builds/ngap/beslog2json.py:182]: looks like `send_it=False` by default within the scope of `request_log_to_json`. How about:
```python
def request_log_to_json(log_fields, json_log_line, send_it=False):
```

File: builds/ngap/beslog2json.py [hunk 4]
```diff
@@ -0,0 +1,560 @@
...
+    debug("REQUEST_MESSAGE_TYPE")
+    
+    send_it = False
+    if TRANSMIT_REQUEST_LOG:
+        json_log_line[CLIENT_IP_KEY] = log_fields[4]
```
Comment (Mikejmnez) [inline builds/ngap/beslog2json.py:187]: add here that if TRANSMIT_REQUEST_LOG then send_it=True:
```python
if TRANSMIT_REQUEST_LOG:
    send_it=True
    ...
```

File: builds/ngap/beslog2json.py [hunk 5]
```diff
@@ -0,0 +1,560 @@
...
+        
+    return send_it
+# - - - - - - - - - - - - - - - - - - - - - - - - - - - - 
+
+
+##########################################################
+# info_log_to_json()
+# Ingest a BES info log record.
+#
+def info_log_to_json(log_fields, json_log_line):
```
Comment (Mikejmnez) [inline builds/ngap/beslog2json.py:216]:
```python
def info_log_to_json(log_fields, json_log_line, send_it=False):
    """docstring here
    """
    ...
    if TRANSMIT_INFO_LOG:
        send_it = True
```

File: builds/ngap/beslog2json.py [hunk 6]
```diff
@@ -0,0 +1,560 @@
...
+    send_it = False
+    log_fields = log_line.split(bes_square_bracket_log_delimiter)
+
+    # Remove the leadin "["  from the time string.
+    time_str = log_fields[0]
+    if time_str.startswith("["):
+        time_str = time_str[1:]
+    json_log_line[TIME_KEY] = time_str
+
+    pid = log_fields[1]
```
Comment (Mikejmnez) [inline builds/ngap/beslog2json.py:369]: refactor perhaps to :
```python
pid = log_fields[1][4:] # perhaps comment here why start at 4?
```

**Review by Mikejmnez — APPROVED** [review]
On the python side of things - very small stylistic suggestions that will reduce size of code.

File: builds/ngap/beslog2json.py (hunk 1 above)
Comment (ndp-opendap) [reply builds/ngap/beslog2json.py:62]: but... How is this useful to make the descriptions "data" that cannot be accessed? What is this?

Comment (Mikejmnez) [reply builds/ngap/beslog2json.py:62]: Are the descriptions (docstring) to debug()  (```Writes "msg" to stderr when the debug_flag is enabled```) being accessed by another function/script? My comment/suggestion has to do with a standardized way to define a function via [their docstrings](https://peps.python.org/pep-0257/)

**Review by jgallagher59701 — APPROVED** [review]
Ready to merge!

**Pull request merged**